library/src/clj_lambda/aws.clj

```diff
@@ -109,8 +110,15 @@
         (store-jar-to-bucket (File. jar-file)
                              bucket
                              object-key)
-        (create-lambda-fn (-> env-settings
-                              (select-keys [:function-name :handler :timeout
-                                            :environment :memory-size :region])
-                              (assoc :role-arn role-arn :bucket bucket :object-key object-key))))
+        ; There seems to be a race condition in the Amazon API which can cause function creation
+        ; to fail if the role used has only recently been created. So retry until this succeeds.
+        ; See:
+        ;   https://stackoverflow.com/questions/36419442/the-role-defined-for-the-function-cannot-be-assumed-by-lambda
+        ;   https://stackoverflow.com/questions/37503075/invalidparametervalueexception-the-role-defined-for-the-function-cannot-be-assu
+        (try-try-again 
+          {:decay :exponential}
```

Comment:
I definitely agree with retrying this operation. However, I think the retry options should be tuned a bit. We should not retry forever. For example there is no check if user specifies correct parameters (mem size etc.) to Lambda function. In that case operation runs in retry loop forever. 

Also default sleep timeout seems to be in 10 seconds which is quite long for first retry (and library writer recommends to update that one). When I have experienced the race condition it usually takes only second or two. 

So what about having a retry settings like this: 3 retries, 1 second sleep with decay?